OpenBikeSensorFirmware/configServer.cpp

```diff
@@ -1065,6 +1069,90 @@ void startServer() {
 
   server.on("/privacy_action", privacyAction);
 
+  // ###############################################################
+  // SD card file systen access
+  // ###############################################################
+
+  server.on("/sd", []() {
+    String path = "/";
+    if (server.hasArg("path")) {
+      path = server.arg("path");
+    }
+
+    File file = SDFileSystem.open(path);
+
+    if (!file) {
+      server.send(404, "text/plain", "File not found.");
+      return;
+    }
+
+
+    if (file.isDirectory()) {
+      String html = header;
+
+      // Header
+      html.replace("{version}", OBSVersion);
+      html.replace("{subtitle}", "SD Card Contents");
+
+      html += "<ul class=\"directory-listing\">";
+
+      // Iterate over directories
+      File child = file.openNextFile();
+      while(child) {
+        html += ("<li class=\""
+          + String(child.isDirectory() ? "directory" : "file")
+          + "\"><a href=\"/sd?path="
+          + String(child.name())
+          + "\">"
+          + String(child.name()).substring(1)
+          + String(child.isDirectory() ? "/" : "")
+          + "</a></li>");
+
+        child.close();
+        child = file.openNextFile();
+      }
+
+      file.close();
+
+      html += "</ul>";
+      html += footer;
+      server.send(200, "text/html", html);
+
+      file.close();
```

Comment:
Is already closed 4 lines above

Reply:
Yeah I should remove that. I tried reopening the file and only reading directories the first time, then files the second time, to get them sorted. But that somehow did not work. Forgot to remove this line from the second time.